ci-tests/Cargo.toml

```diff
@@ -0,0 +1,12 @@
+[package]
```

Comment:
What's the rationale for adding a new crate for this? Why should we run these tests in CI only?

If we're going to put these tests into a separate crate, I think we should probably just rename `connect-tests` to `ci-tests`.

Reply:
The tests are not strictly ci-only 😅, we can run them on dev machines as well. The point was to not break rustls tests on dev machines in case OpenSSL 3 is not installed.

As for combining them with `connect-tests`, I thought since those tests require network connectivity and the tests in `ci-tests` don't, it may be good to keep them separate. I can combine them if that is preferred.

Reply:
I think I would prefer still collecting those in a single crate. @ctz / @cpu thoughts?

Reply:
I think I would also prefer a single crate.

Reply:
I will combine them in a single crate, `ci-tests`.

Reply:
So I tired doing this. A couple of issues:
- `connect-tests` run daily, which may or may not be an issue.
- `connect-tests` run on multiple platforms, including Windows and Mac, and my `ci-tests` required OpenSSL. By running on Ubuntu only, I can use OpenSSL that's bundled with Ubuntu. If we were to run these tests on other platforms as well, we'd need to install OpenSSL on those platforms before running the tests.

Reply:
Okay, that's good feedback. In light of this, maybe we should keep them as two crates after all. I'd suggest you rename the new crate to openssl-tests to be more explicit about what it provides.

Reply:
Yes, that sounds good to me too. #423 sounds like it could reuse that in the future.

Reply:
Done!